runtime/onert/api/nnfw/include/nnfw_experimental.h

```diff
@@ -374,6 +374,28 @@ NNFW_STATUS nnfw_train_input_tensorinfo(nnfw_session *session, uint32_t index,
 NNFW_STATUS nnfw_train_expected_tensorinfo(nnfw_session *session, uint32_t index,
                                            nnfw_tensorinfo *info);
 
+/**
+ * @brief Disable update of the node indicated by op_index during training
+ * @note  This function should be called after {@link nnfw_train_prepare}.
+ *        The update of all the nodes is enabled by default.
+ *
+ * @param[in]   session    The session with loaded model whose node update should be disabled
+ * @param[in]   op_index   The index of training model node
+ * @return @c NNFW_STATUS_NO_ERROR if successful
+ */
+NNFW_STATUS nnfw_train_disable_node_update(nnfw_session *session, uint32_t op_index);
```

Comment:
How about introducing the `_trainable_ops_idx` to the `nnfw_train_info` structure? If so, there is no need to introduce these APIs. The API introduced is not easy to change. And the `_trainable_ops_idx` data is from metadata. So it is better to handle this attribute at the same position as any other attributes in the metadata.

Reply:
Just for your information, 

If doing so, instead of introducing a new API, you could modify `nnfw_train_set_traininfo` and `nnfw_train_get_traininfo `.

https://github.com/Samsung/ONE/blob/9bb7de7bc3f60ae9a92978d1791314a947a9945e/runtime/onert/api/nnfw/include/nnfw_experimental.h#L225-L250

Reply:
I see, it makes a sense for me. Moved to `nnfw_train_info`. I am only not sure about maximum number of trainable ops. Currently I've set `NNFW_TRAINABLE_OPS_MAX_SIZE (256)`. I am rather trying to avoid a dynamic allocation here.

Reply:
I'm also not sure about maximum number of trainable ops, and I'm worried about it always taking as much memory as the maximum. However, I'm not sure if it's OK for us to provide dynamic allocation.
I think we can consider "setting NNFW_TRAINABLE_OPS_MAX_SIZE (256) and using array vs providing dynamic allocation". So I suggests leaving this discussion as a known issue.

Reply:
@ragmani I've created such issue in https://github.com/Samsung/ONE/issues/13140